Delete logic changes in cray-product-catalog, removing from main and sub cm for a product version (#271)
Cray-HPE/cray-product-catalog: CHANGELOG.md

## [Unreleased]

### Added

> - CASM-4368: Delete cray-product-catalog details from main and sub cm for a particular product version
> - Added new argument max_attempts to modify_config_map function in catalog_delete.py as we need not 
>   retry 100 times when read configmap fails for product configmap

Reviewer: ```suggestion
- CASM-4368: Delete Cray Product Catalog details from main and sub ConfigMaps for a particular
  product version.
- Added new argument `max_attempts` to `modify_config_map` function in
  [`catalog_delete.py`](cray_product_catalog/catalog_delete.py), because we need not retry 100
  times when read ConfigMap fails for a product ConfigMap.
```
